core/src/gl/texture.h

```diff
@@ -56,48 +56,42 @@ class Texture {
 
     Texture(const uint8_t* data, size_t length, TextureOptions _options);
 
-    Texture(Texture&& _other) noexcept;
-    Texture& operator=(Texture&& _other) noexcept;
+    Texture(const Texture& _other) = delete;
+    Texture& operator=(const Texture& _other) = delete;
+    Texture(Texture&& _other) = delete;
+    Texture& operator=(Texture&& _other) = delete;
 
     virtual ~Texture();
 
     bool loadImageFromMemory(const uint8_t* data, size_t length);
 
     /* Sets texture pixel data */
-    void setPixelData(const GLuint* data, size_t length);
+    bool movePixelData(int _width, int _height, int _bytesPerPixel, GLubyte* _data, size_t _length);
+    bool setPixelData(int _width, int _height, int _bytesPerPixel, const GLubyte* _data, size_t _length);
 
     void setRowsDirty(int start, int count);
 
     void setSpriteAtlas(std::unique_ptr<SpriteAtlas> sprites);
 
-    /* Perform texture updates, should be called at least once and after adding data or resizing */
-    virtual void update(RenderState& rs, GLuint _textureSlot);
-
-    virtual void update(RenderState& rs, GLuint _textureSlot, const GLuint* data);
-
     /* Resize the texture */
     void resize(int width, int height);
 
-    void bind(RenderState& rs, GLuint _unit);
+    bool bind(RenderState& rs, GLuint _unit);
 
     /* Width and Height texture getters */
-    int getWidth() const { return m_width; }
-    int getHeight() const { return m_height; }
+    int width() const { return m_width; }
```

Comment:
lol I _just_ renamed these to include "get" https://github.com/tangrams/tangram-es/pull/1930

What's your thought process here?

Reply:
I thought we decided a while ago to get rid of these get-prefix for basic getters and this was old leftovers - at least it appeared to me like it and was always on my list to get this naming consistent for Texture. It's a simple command like request to get a property it already holds. When there is no work involved in getting things this is not needed to mention. E.g. a surgeon doesn't ask his assistant if he could be so kind to give him a scalpel - it's just 'scalpel' everybody knows in this context what is wanted. If more work is involved I would agree that is inappropriate when the surgeon to just called 'pizza!' while operating.

Reply:
💉